integration/experimental/v3_share_service_command_test.go

```diff
@@ -0,0 +1,297 @@
+		})
+
+		Context("when there is no space set", func() {
+			BeforeEach(func() {
+				helpers.LogoutCF()
+				helpers.LoginCF()
+				helpers.TargetOrg(ReadOnlyOrg)
+			})
+
+			It("fails with no space targeted error message", func() {
+				session := helpers.CF("v3-share-service", serviceInstance, "-s", sharedToSpaceName)
+				Eventually(session.Out).Should(Say("FAILED"))
+				Eventually(session.Err).Should(Say("No space targeted, use 'cf target -s SPACE' to target a space\\."))
+				Eventually(session).Should(Exit(1))
+			})
+		})
+	})
+
+	Context("when the environment is set up correctly", func() {
+		var (
+			domain      string
+			service     string
+			servicePlan string
+		)
+
+		BeforeEach(func() {
+			service = helpers.PrefixedRandomName("SERVICE")
+			servicePlan = helpers.PrefixedRandomName("SERVICE-PLAN")
+
+			helpers.CreateOrgAndSpace(sharedToOrgName, sharedToSpaceName)
+			setupCF(sourceOrgName, sourceSpaceName)
+
+			domain = defaultSharedDomain()
+		})
+
+		AfterEach(func() {
+			helpers.QuickDeleteOrg(sourceOrgName)
+			helpers.QuickDeleteOrg(sharedToOrgName)
+		})
+
+		Context("when there is a managed service instance in my current targeted space", func() {
+			var broker helpers.ServiceBroker
+
+			BeforeEach(func() {
+				broker = helpers.NewServiceBroker(helpers.NewServiceBrokerName(), helpers.NewAssets().ServiceBroker, domain, service, servicePlan)
+				broker.Push()
+				broker.Configure()
+				broker.Create()
+
+				Eventually(helpers.CF("enable-service-access", service)).Should(Exit(0))
+				Eventually(helpers.CF("create-service", service, servicePlan, serviceInstance)).Should(Exit(0))
+			})
+
+			AfterEach(func() {
+				broker.Destroy()
+			})
+
+			Context("when I want to shared my service instance to a space in another org", func() {
+				It("shares the service instance from my targeted space with the share-to org/space", func() {
+					session := helpers.CF("v3-share-service", serviceInstance, "-s", sharedToSpaceName, "-o", sharedToOrgName)
+					Eventually(session).Should(Exit(0))
+				})
+			})
+
+			Context("when I want to share my service instance into another space in my targeted org", func() {
+				BeforeEach(func() {
+					helpers.CreateSpace(sharedToSpaceName)
+				})
+
+				It("shares the service instance from my targeted space with the share-to space", func() {
+					session := helpers.CF("v3-share-service", serviceInstance, "-s", sharedToSpaceName)
+					Eventually(session).Should(Exit(0))
+				})
+			})
+
+			Context("when the org I want to share into does not exist", func() {
+				It("fails with an org not found error", func() {
+					session := helpers.CF("v3-share-service", serviceInstance, "-s", sharedToSpaceName, "-o", "missing-org")
+					Eventually(session).Should(Say("FAILED"))
+					Eventually(session.Err).Should(Say("Organization 'missing-org' not found"))
+					Eventually(session).Should(Exit(1))
+				})
+			})
+
+			Context("when the space I want to share into does not exist", func() {
```

Comment:
this is a question and not a suggestion, is this the same scenario as when the user does not have authorization for the share-to space?

Reply:
Well, it depends if the user has no read access or if they have read-but-not-write access.

For the no-read-access case, the behavior is the same as if the space doesn't exist at all. The API call we make is `GET /v3/spaces?names=<shared-to-space-name>&organization_guids=<org-guid>`, and the result is the same (an empty list) whether it doesn't exist or you don't have permission to view.

For the read-but-not-write-access case, we fail later during the API call to actually share the service instance into that space. The error the user sees in this case is
```
You are not authorized to perform the requested action
FAILED
```

Should we try to add a test for the second case?

Reply:
yes please add an integration test for the second case thanks!

Reply:
Great, we've added an integration test for this now :)